### DISCUSSION TITLE
Using boot element inside Setup() method.

### DISCUSSION THREAD

**Discussion by gaguerrero** [quasarframework/quasar#9959] [General - Components / Directives / etc] [answered]
Hi, I'm trying to learn javascript and quasar,
I have created an object of boot type defined as "app.config.globalProperties.$Lib", how can I use it inside setup () method using  composition api? thanks in advance!

**Comment by hawkeye64** [comment]
@gaguerrero There's a couple of ways. The normal JavaScript way would be to just import the file that exports what you want.

A more unconventional way with Vue, is to:

```js
import { getCurrentInstance } from 'vue'

setup () {
  const vm = getCurrentInstance()
  const lib = vm.proxy.$Lib
  // ...
}
```

**Reply by gaguerrero** [reply]
Thank you!

**Comment by rstoenescu** [comment, marked as answer]
The proper way: you need to switch to using the provide/inject mechanism, especially with Composition API.
You can still create a boot file and can `app.provide('your-key', ...your_value....)` then `const x = inject('your-key')` in any component's `setup()`

**Reply by hawkeye64** [reply]
Yep, I forgot about that one...

**Reply by gaguerrero** [reply]
Great! thank you razvan and jeff!
